pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -333,7 +366,37 @@ func TestResponseRoundtrip(t *testing.T) {
 	}
 }
 
+func findHistogramMatchingLabels(metrics util.MetricFamilyMap, name string, labelValuePairs ...string) (*dto.Histogram, error) {
+	metricFamily, ok := metrics[name]
+	if !ok {
+		return nil, fmt.Errorf("no metric with name %v found", name)
+	}
+
+	l := labels.FromStrings(labelValuePairs...)
+	var matchingMetrics []*dto.Metric
+
+	for _, metric := range metricFamily.Metric {
+		if util.MatchesSelectors(metric, l) {
+			matchingMetrics = append(matchingMetrics, metric)
```

Comment:
(very, very much a nit) rather than keeping an array here, could just have a bool to see if one is already found and return if so.

Reply:
I could be convinced either way on this... I have a slight preference for the array as it means we can include the number of matching metrics in the failure message if there are many matching metrics.

Reply:
No need to change it imo.